File: survival-client/src/main/java/net/survival/client/input/Mouse.java
package net.survival.client.input;

public final class Mouse
{
    static double x;
    static double y;
    static double prevX;
    static double prevY;
    static boolean lmbDown;
    static boolean rmbDown;
    static boolean prevLmbDown;
    static boolean prevRmbDown;

    private Mouse() {}

    public static double getX() {
        return x;
    }

    public static double getY() {
        return y;
    }

    public static double getDeltaX() {
        return x - prevX;
    }

    public static double getDeltaY() {
        return y - prevY;
    }

    public static boolean isLmbDown() {
        return lmbDown;
    }

    public static boolean isLmbUp() {
        return !lmbDown;
    }

    public static boolean isLmbPressed() {
        return lmbDown && !prevLmbDown;
    }

    public static boolean isLmbReleased() {
        return !lmbDown && prevLmbDown;
    }

    public static boolean isRmbDown() {
        return rmbDown;
    }

    public static boolean isRmbUp() {
        return !rmbDown;
    }

    public static boolean isRmbPressed() {
        return rmbDown && !prevRmbDown;
    }

    public static boolean isRmbReleased() {
        return !rmbDown && prevRmbDown;
    }
}

File: survival-client/src/main/java/net/survival/client/input/Mouse.kt
package net.survival.client.input

object Mouse {
    const val MODE_CENTERED = 0
    const val MODE_NORMAL = 1

    var x = 0.0
    var y = 0.0
    var prevX = 0.0
    var prevY = 0.0
    var isLmbDown = false
    var isRmbDown = false
    var prevLmbDown = false
    var prevRmbDown = false
    var mode = 0; private set
    var modeChanged = true

    val deltaX: Double get() = x - prevX
    val deltaY: Double get() = y - prevY
    val isLmbUp: Boolean get() = !isLmbDown
    val isLmbPressed: Boolean get() = isLmbDown && !prevLmbDown
    val isLmbReleased: Boolean get() = !isLmbDown && prevLmbDown
    val isRmbUp: Boolean get() = !isRmbDown
    val isRmbPressed: Boolean get() = isRmbDown && !prevRmbDown
    val isRmbReleased: Boolean get() = !isRmbDown && prevRmbDown

    fun changeMode(value: Int) {
        mode = value
        modeChanged = true
    }
}